Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -252,9 +263,18 @@ extension TabScrollingController: UIScrollViewDelegate {
 
         suppressToolbarHiding = false
     }
-
+    
+    func scrollViewDidEndDecelerating(scrollView: UIScrollView) {
+        showOrHideWebViewContainerToolbar()
+    }
```

Comment:
Instead of using scrollViewDidEndDecelerating, you might be better add this to the `scrollViewWillEndDragging(_:withVelocity:targetContentOffset:)` method to know where the user will end up when the scrolling is finished. This way you can know before hand if we need to animate in the toolbar. As far as actually animating in the toolbar at the right time to make it look seemless, that might be a bit tricky but I think with the velocity you'll be able to calculate the time at which the top of the scroll view will appear.